llvm/lib/Transforms/Scalar/LowerMatrixIntrinsics.cpp

```diff
@@ -2177,6 +2209,31 @@ class LowerMatrixIntrinsics {
                      Builder);
   }
 
+  /// Lower cast instructions.
+  void VisitCastInstruction(CastInst *Inst, const ShapeInfo &Shape) {
+    Value *Op = Inst->getOperand(0);
+
+    IRBuilder<> Builder(Inst);
+
+    MatrixTy Result;
+    MatrixTy M = getMatrix(Op, Shape, Builder);
+
+    Builder.setFastMathFlags(getFastMathFlags(Inst));
+
+    auto *OrigVTy = cast<VectorType>(Inst->getType());
+    auto *NewVTy = VectorType::get(OrigVTy->getElementType(),
+                                   ElementCount::getFixed(M.getStride()));
+
+    for (unsigned I = 0; I < Shape.getNumVectors(); ++I)
```

Comment:
```suggestion
    for (auto &Vector : M.vectors()) {
```

Should be able to use `M.vectors()` here